src/psyclone/domain/lfric/lfric_builtins.py

```diff
@@ -310,10 +311,25 @@ def reference_element(self):
         '''
         return None
 
-    @abc.abstractmethod
     def gen_code(self, parent):
-        ''' Must be implemented in sub-class. Will generate the f2pygen AST
-        for this builtin. '''
+        '''
+        Generates LFRic API specific PSy code (in the form of f2pygen AST
+        nodes) for a call to this Built-in. This method must be overridden
+        if a sub-class does not yet implement lower_to_language_level().
+
+        :param parent: Node in f2pygen tree to which to add call.
+        :type parent: :py:class:`psyclone.f2pygen.BaseGen`
+
+        '''
+        # Ultimately this routine will be removed once the LFRic PSyIR
+        # has been fully migrated to use the PSyIR backends (e.g. #1010).
+        # For now we create a temporary copy of the PSyIR, lower it and
+        # then create an f2pygen node from it.
+        routine = Routine('dummy')
+        new_tree = self.copy()
+        routine.addchild(new_tree)
+        new_tree.lower_to_language_level()
```

Comment:
There is the possibility that you don't need the lowering statement here anymore, but I am not entirely sure.

Reply:
Oh yes, it seems we don't :-)